src/java.base/share/classes/java/lang/ClassLoader.java

```diff
@@ -1685,6 +1685,19 @@ public InputStream getResourceAsStream(String name) {
         }
     }
 
+    /**
+     * Called by VM for reading class bytes.
+     */
+    private byte[] getResourceAsByteArray(String name) {
+        Objects.requireNonNull(name);
+        InputStream is = getResourceAsStream(name);
+        try {
+            return is != null ? is.readAllBytes() : null;
+        } catch (IOException e) {
+            return null;
+        }
```

Comment:
Any `IOException` should just be left pending so that it gets reported eventually. Otherwise you are returning null here but asserting in the VM that null is never returned.

Reply:
I've changed the java method to throw IOException and removed the try-catch statements.